csharp/src/Google.Protobuf/CodedInputStream.cs

```diff
@@ -577,7 +577,32 @@ public uint ReadFixed32()
         /// </summary>
         public bool ReadBool()
         {
-            return ReadRawVarint32() != 0;
+            int result = 0; // can't do AND on bytes, it gets implicitly upcasted to int

```

Comment:
good catch but the fix seems overly complex, just `ReadRawVarint64() != 0` should be enough.

FTR that is also what Java does.
https://github.com/protocolbuffers/protobuf/blob/3cae867740e409d9b9fbc1224c35357a49ce0ce2/java/core/src/main/java/com/google/protobuf/CodedInputStream.java#L782

Reply:
Right, I thought about that. Though I also thought that maybe reading an actual varint would be excessive work. The case 99% of the time is that we have at least one byte left in the buffer and the varint is a 1 or 0. ReadRawVarint64 needs at least 10 bytes and does bit shifts and other things just to ultimately check if the whole value has at least one bit set.